leftwm-core/src/handlers/window_handler.rs

```diff
@@ -56,6 +56,10 @@ impl<C: Config, SERVER: DisplayServer> Manager<C, SERVER> {
             self.state.focus_window(&window.handle);
         }
 
+        if self.config.move_new_windows_to_top() {
+            self.state.move_to_top(&window.handle);
+        }
+
```

Comment:
Rather than handling this here I would place it in insert_window. Also calling move_to_top may cause the tiled window to appear above floating windows, instead I would insert the window first in the workspace before the sorting happens. Placing your check in this block of insert_window should work:
```
        if matches!(layout, Layout::Monocle | Layout::MainAndDeck) || new_option { // Add check for new option
            // Extract the current windows on the same workspace.
            let mut to_reorder = helpers::vec_extract(&mut state.windows, for_active_workspace);
            if layout == Layout::Monocle || to_reorder.is_empty() || new_option { // Add check for new option
                // When in monocle we want the new window to be fullscreen if a window was
                // fullscreen.
                if was_fullscreen {
                    let act = DisplayAction::SetState(
                        window.handle,
                        !window.is_fullscreen(),
                        WindowState::Fullscreen,
                    );
                    state.actions.push_back(act);
                }
                // Place the window above the other windows on the workspace.
                to_reorder.insert(0, window.clone());
            } else {
```

Reply:
Cool I will improve it later today, thanks!

Reply:
Unfortunately, `insert_window` is not aware of config, so the config value I introduced would have to passed in somehow, via an argument maybe?

Also, shouldn't it be placed just above this?:
```
    // Past special cases we just push the winodw to the bottom.
    state.windows.push(window.clone());
```

Reply:
Insertion would be done like this `state.windows.insert(0, window.clone());` I imagine